### PR TITLE
refacored the keys tab removed duplication

### PR DESCRIPTION
**Pull request by edewit**

### CODE REVIEW

**Review by jonkoops — CHANGES_REQUESTED** [review]
@edewit seems there is a merge conflict, can you rebase your work?

**Comment by edewit** [comment]
done

File: src/realm-settings/routes/KeyProvider.ts [hunk 1]
```diff
@@ -4,21 +4,20 @@ import { generatePath } from "react-router-dom";
 import type { RouteDef } from "../../route-config";
 
 export type KeyProviderParams = {
-  realm: string;
   id: string;
+  providerType: string;
```
Comment (jonkoops) [inline src/realm-settings/routes/KeyProvider.ts:8]: If this is a set of limited options you could create a special type for it.

File: src/realm-settings/messages.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ export default {
       "If you delete this realm, all associated data will be removed.",
     deleteProviderTitle: "Delete key provider?",
     deleteProviderConfirm:
-      "Are you sure you want to permanently delete the key provider ",
+      "Are you sure you want to permanently delete the key provider {{provider}} ?",
```
Comment (jonkoops) [inline src/realm-settings/messages.ts:13]:
```suggestion
      "Are you sure you want to permanently delete the key provider {{provider}}?",
```

File: src/realm-settings/keys/KeysListTab.tsx [hunk 3]
```diff
@@ -32,18 +35,17 @@ type KeysListTabProps = {
...
+  const { t } = useTranslation("realm-settings");
   const history = useHistory();
-  const { url } = useRouteMatch();
 
   const [key, setKey] = useState(0);
   const [publicKey, setPublicKey] = useState("");
   const [certificate, setCertificate] = useState("");
   const [filterDropdownOpen, setFilterDropdownOpen] = useState(false);
-  const [filterType, setFilterType] = useState("Active keys");
+  const [filterType, setFilterType] = useState<string>(FILTER_OPTIONS[0]);
```
Comment (jonkoops) [inline src/realm-settings/keys/KeysListTab.tsx:45]: I think you can omit the type here, as it should be inferrable by the default you are providing.
```suggestion
  const [filterType, setFilterType] = useState(FILTER_OPTIONS[0]);
```

File: src/realm-settings/keys/KeysListTab.tsx [hunk 4]
```diff
@@ -55,81 +57,35 @@ export const KeysListTab = ({ realmComponents }: KeysListTabProps) => {
     });
 
     const keys = keysMetaData.keys;
+    const filtered =
+      filterType !== FILTER_OPTIONS[0]
+        ? keys?.filter((i) => i.status === filterType)
```
Comment (jonkoops) [inline src/realm-settings/keys/KeysListTab.tsx:62]: You can get rid of the single letter variable if you use destructuring here.
```suggestion
        ? keys?.filter(({ status }) => status === filterType)
```

**Review by jonkoops — CHANGES_REQUESTED** [review]
This PR is amazing Erik, love the cleanup!

File: cypress/support/pages/admin_console/manage/realm_settings/KeysTab.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  goToKeysTab() {
+    cy.findByTestId(this.keysTab).click();
+
+    return this;
+  }
+
+  goToProvidersTab() {
+    this.goToKeysTab();
```
[the remark this answers is not in the document]
Comment (edewit) [reply cypress/support/pages/admin_console/manage/realm_settings/KeysTab.ts:13]: yes it is

File: src/clients/authorization/AuthorizationEvaluate.tsx [hunk 6]
```diff
@@ -189,7 +189,7 @@ export const AuthorizationEvaluate = ({
     }
   };
 
-  const handleKeyDown = (e: any) => {
+  const handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
```
[the remark this answers is not in the document]
Comment (edewit) [reply src/clients/authorization/AuthorizationEvaluate.tsx:192]: I know

**Review by jonkoops — APPROVED** [review]

**Comment by jonkoops** [comment]
Looks good to me, @edewit can you verify the tests are still passing locally?

**Comment by edewit** [comment]
```
       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  authentication_policies.spec.ts          00:33        3        3        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  authentication_test.spec.ts              02:57       15       15        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  client_authorization_test.spec.ts        00:33        9        9        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  client_scopes_test.spec.ts               01:20       20       20        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  clients_saml_test.spec.ts                00:28        5        5        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  clients_test.spec.ts                     04:24       25       25        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  events_test.spec.ts                      02:01       14       14        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  group_test.spec.ts                       01:24        9        9        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  identity_providers_oidc_test.spec.t      00:57        5        5        -        -        - │
  │    s                                                                                           │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  identity_providers_saml_test.spec.t      02:20       12       12        -        -        - │
  │    s                                                                                           │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  identity_providers_test.spec.ts          01:29        9        9        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  login_test.spec.ts                       00:11        2        2        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  masthead_test.spec.ts                    00:31        2        2        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  partial_export_test.spec.ts              00:27        3        3        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  partial_import_test.spec.ts              01:06        4        4        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  realm_roles_test.spec.ts                 00:21        7        7        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  realm_settings_client_policies_test      01:02       18       18        -        -        - │
  │    .spec.ts                                                                                    │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  realm_settings_client_profiles_test      01:01       21       21        -        -        - │
  │    .spec.ts                                                                                    │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  realm_settings_events_test.spec.ts       02:52       17       16        -        1        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  realm_settings_tabs_test.spec.ts         00:57        6        6        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  realm_settings_user_profile_tab.spe      00:30        3        3        -        -        - │
  │    c.ts                                                                                        │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  realm_test.spec.ts                       00:22        4        4        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  realm_user_registration.spec.ts          00:18        3        3        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  sessions_test.spec.ts                    00:41        5        5        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  user_fed_kerberos_test.spec.ts           01:26        9        9        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  user_fed_ldap_hardcoded_mapper_test      02:24       12       12        -        -        - │
  │    .spec.ts                                                                                    │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  user_fed_ldap_mapper_test.spec.ts        02:27       12       12        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  user_fed_ldap_test.spec.ts               01:38        8        8        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  users_test.spec.ts                       00:38       13       12        -        1        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        37:33      275      273        -        2        -  
```

**Pull request merged**